doc/user-manual/language/cubical.lagda.rst

```diff
@@ -97,10 +158,780 @@ of the interval ``i0`` and ``i1`` are the bottom and top elements, respectively
+
+::
+
+  hfill : ∀ {ℓ} {A : Set ℓ} {φ : I}
+          (u : ∀ i → Partial φ A) (u0 : A [ φ ↦ u i0 ])
+          (i : I) → A
+  hfill {φ = φ} u u0 i = hcomp (λ j → λ { (φ = i1) → u (i ∧ j) 1=1
+                                        ; (i = i0) → ouc u0 })
+                               (ouc u0)
+
+When ``i`` is ``i0`` this is ``u0`` and when ``i`` is ``i1`` this is
+``hcomp u u0``. This can hence be seen as giving us the interior of an
+open box. In the special case of the square above ``hfill`` gives us a
+direct cubical proof that composing ``p`` with ``refl`` is ``p``.
+
+::
+
+  compPathRefl : ∀ {ℓ} {A : Set ℓ} {x y : A} (p : x ≡ y) → compPath p refl ≡ p
+  compPathRefl {x = x} {y = y} p j i = hfill (λ _ → λ { (i = i0) → x
+                                                      ; (i = i1) → y })
+                                             (inc (p i))
+                                             (~ j)
+
+
+Glue types
 ----------
+
+In order to be able to prove the univalence theorem we also have to
+add "Glue" types. These lets us turn equivalences between types into
+paths between types. An equivalence of types ``A`` and ``B`` is
+defined as a map ``f : A → B`` such that its fibers are contractible.
+
+.. code-block:: agda
+
+  fiber : ∀ {ℓ} {A B : Set ℓ} (f : A → B) (y : B) → Set ℓ
+  fiber {A = A} f y = Σ[ x ∈ A ] f x ≡ y
+
+  isContr : ∀ {ℓ} → Set ℓ → Set ℓ
+  isContr A = Σ[ x ∈ A ] (∀ y → x ≡ y)
+
+  record isEquiv {ℓ} {A B : Set ℓ} (f : A → B) : Set ℓ where
+    field
+      equiv-proof : (y : B) → isContr (fiber f y)
+
+  _≃_ : ∀ {ℓ} (A B : Set ℓ) → Set ℓ
+  A ≃ B = Σ[ f ∈ (A → B) ] (isEquiv f)
+
+The simplest example of an equivalence is the identity function.
+
+::
+
+  idfun : ∀ {ℓ} → (A : Set ℓ) → A → A
+  idfun _ x = x
+
+  idIsEquiv : ∀ {ℓ} (A : Set ℓ) → isEquiv (idfun A)
+  equiv-proof (idIsEquiv A) y =
+    ((y , refl) , λ z i → z .snd (~ i) , λ j → z .snd (~ i ∨ j))
+
+  idEquiv : ∀ {ℓ} (A : Set ℓ) → A ≃ A
+  idEquiv A = (idfun A , idIsEquiv A)
+
+
+An important special case of equivalent types are isomorphic types
+(i.e. types with maps going back and forth which are mutually
+inverse): https://github.com/agda/cubical/blob/master/Cubical/Foundations/Isomorphism.agda.
+
+As everything has to work up to higher dimensions the Glue types take
+a partial family of types that are equivalent to the base type ``A``:
+
+.. code-block:: agda
+
+  primGlue : ∀ {ℓ ℓ'} (A : Set ℓ) {φ : I} (T : Partial φ (Set ℓ'))
+           → (e : PartialP φ (λ o → T o ≃ A)) → Set ℓ'
+
+These come with a constructor and eliminator:
+
+.. code-block:: agda
+
+  glue : ∀ {ℓ ℓ'} {A : Set ℓ} {φ : I} {T : Partial φ (Set ℓ')}
+           {e : PartialP φ (λ o → T o ≃ A)}
+         → PartialP φ T → A → primGlue A T e
+
+  unglue : ∀ {ℓ ℓ'} {A : Set ℓ} (φ : I) {T : Partial φ (Set ℓ')}
+             {e : PartialP φ (λ o → T o ≃ A)} → primGlue A T e → A
+
+In the cubical library we uncurry the Glue types to make them more
+pleasant to use:
+
```

Comment:
Given that you expect the users to use agda/cubical, it is rather irrelevant here that `Glue` and `primGlue` are different, you should talk only about `Glue` and mention the difference in currification in the appendix.

Reply:
Good point! I didn't like it when I wrote this...